vm/src/org/mozartoz/truffle/nodes/builtins/NumberBuiltins.java

```diff
@@ -61,6 +64,52 @@ long neg(long value) {
 
 	}
 
+	@Builtin(name = "pow", deref = ALL)
+	@GenerateNodeFactory
+	@NodeChildren({ @NodeChild("left"), @NodeChild("right") })
+	public static abstract class PowNode extends OzNode {
+
+		@Specialization(guards = { "isInteger(base)", "exponent <= 94", "exponent == cachedExponent" })
```

Comment:
Could you explain why 94?
Since this Specialization does not mention a `limit`, it means it uses the default of `3` cachedExponent, after that it uses the one below.